flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/ExecutionVertex.java

```diff
@@ -632,8 +632,8 @@ void executionFailed(Throwable t) {
 	/**
 	 * Simply forward this notification. This is for logs and event archivers.
 	 */
-	void notifyStateTransition(ExecutionAttemptID executionId, ExecutionState newState, Throwable error) {
-		getExecutionGraph().notifyExecutionChange(getJobvertexId(), subTaskIndex, executionId, newState, error);
+	void notifyStateTransition(ExecutionAttemptID executionId, ExecutionState newState, Throwable error, long errorTimestamp) {
```

Comment:
Technically, yes, it is the TS when the Execution went into the state FAILED.

Since we can't use the TM timestamp this seemed like the most appropriate one to take. The only other option would be to take a TS when the UpdateTaskExecutionState message hits the JM; at that point though we are already at a point where the timestamp isn't particularly accurate, so by using the FAILED timestamp we aren't really losing accuracy, but affect fewer methods.